packages/cf-deploy-config-sub-generator/src/app/index.ts

```diff
@@ -156,29 +170,45 @@ export default class extends DeploymentGenerator {
         if (!baseConfigExists) {
             bail(ErrorHandler.noBaseConfig(baseConfigFile));
         }
-
         this.deployConfigExists = this.fs.exists(join(this.appPath, this.options.config ?? FileName.Ui5Yaml));
     }
 
     public async prompting(): Promise<void> {
         if (this.abort) {
             return;
         }
-
-        if (this.isCap && this.projectRoot && !this.mtaPath) {
-            // if the user is adding deploy config to a CAP project and there is no mta.yaml in the root, then log error and exit
-            this.abort = true;
-            handleErrorMessage(this.appWizard, { errorType: ERROR_TYPE.CAP_DEPLOYMENT_NO_MTA });
-            return;
+        if (!this.launchDeployConfigAsSubGenerator) {
+            await this._prompting();
         }
+        await this._reconcileAnswersWithOptions();
+    }
 
-        if (!this.launchDeployConfigAsSubGenerator) {
+    private async _prompting(): Promise<void> {
+        const isCAPMissingMTA = this.isCap && this.projectRoot && !this.mtaPath;
+        if (isCAPMissingMTA) {
+            DeploymentGenerator.logger?.debug(t('cfGen.debug.capMissingMTA'));
+            // If launched as root generator, add a prompt to allow user decide if they want to add an MTA config
+            let questions = (await getCFApprouterQuestionsForCap({
+                projectRoot: this.projectRoot ?? process.cwd()
+            })) as Question[];
+            questions = withCondition(questions, (answers: Answers) => answers.addCapMtaContinue === true);
+            questions.unshift(...getConfirmMtaContinuePrompt());
+            this.appRouterAnswers = (await this.prompt(questions)) as CfAppRouterDeployConfigAnswers;
+            if ((this.appRouterAnswers as Answers).addCapMtaContinue !== true) {
+                this.abort = true;
+                return;
+            }
+            // Configure defaults
+            this.destinationName = DefaultMTADestination;
+            this.options.overwrite = true; // Don't prompt the user to overwrite files we've just written!
+            this.answers = {};
+            this.answers.destinationName = this.destinationName;
+            this.answers.addManagedAppRouter = false;
+        } else {
```

Comment:
Maybe put all the mta related code in this condition into a new function? But its just a suggestion.

Reply:
Thanks, I will do another pass to clean it up once its merged.